lib/spack/spack/solver/concretize.lp

```diff
@@ -752,10 +752,14 @@ node_has_variant(node(ID, Package), Variant) :-
   pkg_fact(Package, variant(Variant)),
   attr("node", node(ID, Package)).
 
-attr("variant_propagate", PackageNode, Variant, Value, Source) :-
+attr("variant_possible_prop", PackageNode, Variant, Value, Source) :-
   attr("node", PackageNode),
   depends_on(ParentNode, PackageNode),
-  attr("variant_propagate", ParentNode, Variant, Value, Source),
+  attr("variant_value", node(ID, Source), Variant, Value),
```

Comment:
Hmm, here we don't fix the `ID` variable. So what would happen if we have `node(0, Source)` and `node(1, Source` in the same graph? How do they relate to `PackageNode` and `ParentNode`? 

I think the initial source should be a node too, but am unsure what is the exact semantic of propagation. Is there any place where we document that?